brian2/core/variables.py

```diff
@@ -165,6 +164,16 @@ def __init__(self, name, dimensions=DIMENSIONLESS, owner=None, dtype=None,
         #: Whether the variable is an array
         self.array = array
 
+        #: Mark the list of attributes that should not be considered for caching
+        #: of state update code, etc.
+        self._cache_irrelevant_attributes = ['owner']
```

Comment:
Indeed! It would even simplify the code a bit, since it wouldn't appear in `__dict__` itself!